File: data_structures/linked_list/java/samuelrey/LinkedList.java
import java.lang.IndexOutOfBoundsException;

public class LinkedList<T> {

    private Node<T> head;
    private Node<T> tail;
    private int count;

    public LinkedList() {
        head = null;
        tail = null;
        count = 0;
    }

    public void insert(T data) {
        if(count == 0) {
            head = new Node(data);
            tail = head;
        }

        else {
            tail.next = new Node(data);
        }

        count++;
    }

    public void remove(int index) {
        if(index < 0 || index >= count) {
            throw new IndexOutOfBoundsException();
        }

        if(index == 0) {
            if(head == tail) {
                tail = null;
            }
            head = head.next;
        }

        else {
            Node<T> tmp = head;
            while(index > 1) {
                tmp = tmp.next;
                index--;
            }
            if(tmp.next == tail) {
                tmp.next = null;
            }
            else {
                tmp.next = tmp.next.next;
            } 
        }

        count--;
    }

    public T get(int index) {
        if(index < 0 || index >= count) {
            throw new IndexOutOfBoundsException();
        }

        Node<T> tmp = head;
        while(index > 0) {
            tmp = tmp.next;
        }

        return tmp.data;
    }

    private class Node<T> {
        public T data;
        public Node<T> next;

        public Node(T data) {
            this.data = data;
            this.next = null;
        }
    }
}

File: data_structures/linkedlist.py
class Node:
    def __init__(self, data=None):
        self.data=data
        self.next=None #Node's will hold data & pointer to next node

class LinkedList:
    def __init__(self):
        self.head=Node() 
    def append(self, data):
        node = Node(data)
        cur = self.head
        while cur.next != None:
            cur = cur.next
        cur.next=node
    def traverse(self):
        cur = self.head
        while cur.next != None:
            cur = cur.next
            print(cur.data, "->")
    def remove(self, data):
        cur = self.head
        while cur.next != None:
            if cur.data == data:
                cur.next = None

LL = LinkedList()
LL.append(5)
LL.append(6)
LL.append(7)
LL.append(8)
LL.traverse()
